### PR TITLE
feat(platform): anywhere component upgrade

### PR DESCRIPTION
**Pull request by wl-chen**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/platform/controller/cluster/cluster_controller.go [hunk 1]
```diff
@@ -454,7 +453,13 @@ func (c *Controller) onUpdate(ctx context.Context, cluster *platformv1.Cluster)
 				_, _ = c.platformClient.ClusterCredentials().Update(ctx, clusterWrapper.ClusterCredential, metav1.UpdateOptions{})
 			}
 
-			_, _ = c.platformClient.Clusters().UpdateStatus(ctx, clusterWrapper.Cluster, metav1.UpdateOptions{})
+			updatedCluster, _ := c.platformClient.Clusters().UpdateStatus(ctx, clusterWrapper.Cluster, metav1.UpdateOptions{})
+			if c.isCRDMode && cluster.Spec.Type == "Anywhere" {
```
Comment (wl-chen) [inline pkg/platform/controller/cluster/cluster_controller.go:457]: anywhere 删掉

File: pkg/util/extenderapi/extenderapi.go [hunk 2]
```diff
@@ -97,3 +97,24 @@ func GetHelmRelease(clientSet client.Client, name, namespace string) (*appsv1alp
 	}
 	return hr, nil
 }
+
+func ListHelmRelease(clientSet client.Client, namespace string) (*appsv1alpha1.HelmReleaseList, error) {
```
Comment (wl-chen) [inline pkg/util/extenderapi/extenderapi.go:101]: 增加一下筛选，只把anywhere系统组件筛选出来

Comment (wl-chen) [reply pkg/util/extenderapi/extenderapi.go:101]: resourceVersion 0

**Comment by leoryu** [comment]
LGTM

**Review by DanielXLee — APPROVED** [review]

File: cmd/tke-platform-controller/app/webhook/mutate.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			return
+		}
+	} else {
+		log.Errorf("request body nil")
+		http.Error(reponseWriter, "request body nil", http.StatusBadRequest)
+		return
+	}
+
+	contentType := request.Header.Get("Content-Type")
+	if contentType != "application/json" {
```
[the remark this answers is not in the document]
Comment (wl-chen) [reply cmd/tke-platform-controller/app/webhook/mutate.go:39]: 目前只有 application/json 场景，其他场景还没碰到，也没测过，所以没加～

**Review by rootdeep — APPROVED** [review]

**Review by leoryu — APPROVED** [review]

**Pull request merged**